### ISSUE TITLE
Customer name of purchases are being overwritten by subsequent purchases

### ISSUE THREAD

**Issue by muesliq** [easydigitaldownloads/easy-digital-downloads#6043]
When `John Doe` with email address `john@example.org` orders again but mistypes his name as `Jane Doe`, not only his customer account is being updated but also **all logs of previous purchases**.

John's actions retroactively become Jane's. John disappears from the system. Clearly a bug in an ecommerce system.

**Comment by pippinsplugins** [comment]
@muesliq that is intentional.

**Comment by muesliq** [comment]
I didn't think it was an accident but I believe the intention is flawed. An ecommerce system should never retroactively alter records. 

Just imagine: I know your email address, so with a simple purchase I can delete your name from all the previous records of your favorite ecommerce shop (under the assumption that log-in isn't mandatory there). 

That should never be possible.

**Comment by pippinsplugins** [comment]
That's a perfectly valid point. I don't think we can say it's wrong to correct a previous error either though. If someone enters their information incorrectly or changes their info between purchases, that should be corrected.

**Comment by muesliq** [comment]
I agree. But that correction could be logged somewhere. And previous purchase logs could contain the previous name.

**Comment by muesliq** [comment]
And related to that: If an old customer mistypes his email address, I can't correct that as there is an account with the correct email address already. It should be possible to merge accounts.

(In our case this causes even further trouble: We deliver downloads by email, not on the website, as they are being rendered after the purchase. Therefor we can't resend a download email in the above case. At least not via EDD.)

**Comment by pippinsplugins** [comment]
Merging the accounts is a separate issue so let's please leave that portion out of this issue for clarity's sake.

Note: you can manually merge accounts together by re-assigning purchases from the bad email to the account with the correct email.

**Comment by muesliq** [comment]
I am not sure it is necessarily a separate issue: Two names + one email address vs. one name + two email addresses.

Both _could_ be addressed with a common functionality. Just that the merge in one case is triggered by the user and in the other by the admin.
